cypress/e2e/registration.cy.js

```diff
@@ -1,11 +1,202 @@
 /// <reference types='cypress' />
 
-describe('Student Registration page', () => {
-  before(() => {
-    
+const { generateNewUser } = require('../support/genereateNewUser');
+
+describe('Registration form', () => {
+  beforeEach(() => {
+    cy.viewport(550, 750) // viewport to 550px x 750px because the page does not scroll
+    cy.visit('https://demoqa.com/automation-practice-form');
+  });
+
+  it('Check the ability successful registration with using all field', () => {
+    const {
+      firstname,
+      lastname,
+      email,
+      currentAddress,
+      phoneNumber
+    } = generateNewUser();
+
+    cy.findById('firstName').type(firstname);
+    cy.findById('lastName').type(lastname);
+    cy.findById('userEmail').type(email);
+    cy.get('label[for="gender-radio-2"]').click();
+    cy.findById('userNumber').type(phoneNumber);
+    cy.findById('dateOfBirthInput').click();
+    cy.get('.react-datepicker__month-select').select('February');
+    cy.get('.react-datepicker__year-select').select('2010');
+    cy.get('[aria-label="Choose Friday, February 12th, 2010"]').click();
+    cy.findById('subjectsInput').type('English{enter}');
+    cy.get('label[for="hobbies-checkbox-1"]').click();
+    cy.findById('currentAddress').type(currentAddress);
+    cy.findById('stateCity-wrapper').scrollIntoView();
+    cy.findById('state').click().type('NCR{enter}');
+    cy.findById('city').click().type('Delhi{enter}');
+    cy.findById('submit').should('be.visible').click();
+
+    cy.get('.modal-content').should('exist');
+    cy.contains('Thanks for submitting the form').should('exist');
+
+    cy.get('.modal-content')
+      .should('contain',`${firstname} ${lastname}`)
+      .should('contain', email)
+      .should('contain',phoneNumber)
+      .should('contain',`English`)
+      .should('contain', currentAddress);
+  });
+
+  it(`The 'First Name' field must be filled in.`, () => {
+    cy.get('#submit').click();
+
+    cy.findById('firstName')
+      .should('have.css', 'border-color', 'rgb(220, 53, 69)');
+  });
+
+  it(`The 'Last Name' field must be filled in.`, () => {
+    cy.get('#submit').click();
+
+    cy.findById('lastName')
+      .should('have.css', 'border-color', 'rgb(220, 53, 69)');
+  });
+
+  it(`The 'Genger' field must be filled in.`, () => {
+    cy.get('#submit').click();
+
+    cy.get('label[for="gender-radio-1"]')
+      .should('have.css', 'border-color', 'rgb(220, 53, 69)');
+    cy.get('label[for="gender-radio-2"]')
+      .should('have.css', 'border-color', 'rgb(220, 53, 69)');
+    cy.get('label[for="gender-radio-3"]')
+      .should('have.css', 'border-color', 'rgb(220, 53, 69)');
+  });
+
+  it(`The 'Mobile' field must be filled in.`, () => {
+    cy.get('#submit').click();
+
+    cy.findById('userNumber')
+      .should('have.css', 'border-color', 'rgb(220, 53, 69)');
```

Comment:
Good approach, but your test names don't reflect your tests. It would be better to say something like "Required empty ${input} is highlighted after clicking on the [Submit] button"